### PR TITLE
fix(api): bound framework-parsed request bodies

### PR DESCRIPTION
**Pull request by fetw882**
## Why

Better Auth and oRPC currently receive request streams without a byte ceiling before their framework parsers consume them. An unauthenticated auth caller or authenticated abnormal client can therefore keep the API buffering an oversized declared or chunked JSON body. RPC still needs to accept the existing 10 MiB attachment envelope.

## What

- cap Better Auth request bodies at 64 KiB and all RPC bodies at 16 MiB
- reject malformed, oversized declared, and oversized streamed bodies with stable JSON 413 before route parsers run
- cancel over-budget streams and replay accepted chunked bodies to downstream parsers
- keep independently bounded voice and webhook routes outside this middleware
- add regression tests for declared, streamed, exact-boundary, and route-scoping behavior

## Tests

- `pnpm --filter @rakazo/api check`
- `pnpm --filter @rakazo/api test`
- `pnpm check`

## Summary by CodeRabbit

* **Bug Fixes**
  * Authentication and RPC endpoints now enforce request body size limits.
  * Oversized, streamed, or invalidly declared requests are rejected with a `413` response before further processing.
  * Valid requests at or below the configured limits continue to be handled normally.
  * Voice synthesis requests retain their independent size limits.
  * Invalid `Content-Length` values, including fractional, hexadecimal, or empty values, are rejected consistently.

### CODE REVIEW

File: apps/api/src/request-body-limit.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  if (!Number.isSafeInteger(maxSize) || maxSize <= 0) {
+    throw new Error("Request body limit must be a positive integer");
+  }
+  return async (c, next) => {
+    const request = c.req.raw;
+    if (!request.body) return next();
+
+    const contentLength = request.headers.get("content-length");
+    if (contentLength !== null && !request.headers.has("transfer-encoding")) {
+      const declared = Number(contentLength);
```
[the remark this answers is not in the document]
Comment (fetw882) [reply apps/api/src/request-body-limit.ts]: Fixed in d91ebd12. Content-Length now has to match the HTTP ASCII decimal grammar before numeric conversion, so empty, fractional, hexadecimal, non-numeric, negative, and over-limit declarations all reject before the route parser. Added table-driven coverage for the requested empty, 8.0, and 0x8 cases plus non-numeric input; all 266 API tests and API typecheck pass.
